diff --git a/node_modules/@dfinity/candid/lib/cjs/idl.js b/node_modules/@dfinity/candid/lib/cjs/idl.js
index d97a72b..9caed53 100644
--- a/node_modules/@dfinity/candid/lib/cjs/idl.js
+++ b/node_modules/@dfinity/candid/lib/cjs/idl.js
@@ -457,8 +457,8 @@ class FixedIntClass extends PrimitiveType {
         return v.visitFixedInt(this, d);
     }
     covariant(x) {
-        const min = BigInt(2) ** BigInt(this._bits - 1) * BigInt(-1);
-        const max = BigInt(2) ** BigInt(this._bits - 1) - BigInt(1);
+        const min = Math.pow(2, this._bits - 1) * BigInt(-1);
+        const max = Math.pow(2, this._bits - 1) - BigInt(-1);
         if (typeof x === 'bigint') {
             return x >= min && x <= max;
         }
@@ -507,7 +507,7 @@ class FixedNatClass extends PrimitiveType {
         return v.visitFixedNat(this, d);
     }
     covariant(x) {
-        const max = BigInt(2) ** BigInt(this.bits);
+        const max = Math.pow(2, this.bits);
         if (typeof x === 'bigint' && x >= BigInt(0)) {
             return x < max;
         }
