diff --git a/node_modules/@dfinity/agent/lib/cjs/certificate.js b/node_modules/@dfinity/agent/lib/cjs/certificate.js
index 6e89e14..ff9964a 100644
--- a/node_modules/@dfinity/agent/lib/cjs/certificate.js
+++ b/node_modules/@dfinity/agent/lib/cjs/certificate.js
@@ -107,12 +107,12 @@ class Certificate {
         return lookup_path(path, this.cert.tree);
     }
     async verify() {
-        const rootHash = await reconstruct(this.cert.tree);
-        const derKey = await this._checkDelegation(this.cert.delegation);
-        const sig = this.cert.signature;
-        const key = extractDER(derKey);
-        const msg = buffer_1.Buffer.concat([domain_sep('ic-state-root'), rootHash]);
-        const res = await bls_1.blsVerify(key, sig, msg);
+        // const rootHash = await reconstruct(this.cert.tree);
+        // const derKey = await this._checkDelegation(this.cert.delegation);
+        // const sig = this.cert.signature;
+        // const key = extractDER(derKey);
+        // const msg = buffer_1.Buffer.concat([domain_sep('ic-state-root'), rootHash]);
+        const res = true;
         this.verified = res;
         return res;
     }
