diff --git a/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java b/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java
index 284e544..c9d483b 100644
--- a/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java
+++ b/node_modules/react-native-device-info/android/src/main/java/com/learnium/RNDeviceInfo/RNDeviceModule.java
@@ -189,7 +189,6 @@ public class RNDeviceModule extends ReactContextBaseJavaModule {
 
     final Map<String, Object> constants = new HashMap<>();
 
-    constants.put("uniqueId", getUniqueIdSync());
     constants.put("deviceId", Build.BOARD);
     constants.put("bundleId", getReactApplicationContext().getPackageName());
     constants.put("systemName", "Android");
diff --git a/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m b/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m
index 70c9634..0006e05 100644
--- a/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m
+++ b/node_modules/react-native-device-info/ios/RNDeviceInfo/RNDeviceInfo.m
@@ -56,7 +56,6 @@ + (BOOL)requiresMainQueueSetup
 
 - (NSDictionary *)constantsToExport {
     return @{
-         @"uniqueId": [self getUniqueId],
          @"deviceId": [self getDeviceId],
          @"bundleId": [self getBundleId],
          @"systemName": [self getSystemName],
@@ -364,7 +363,7 @@ - (NSString *) getBuildId {
     return self.getBuildId;
 }
 
-- (NSString *) getUniqueId {
+RCT_EXPORT_BLOCKING_SYNCHRONOUS_METHOD(getUniqueIdSync) {
     return [DeviceUID uid];
 }
 
diff --git a/node_modules/react-native-device-info/src/index.ts b/node_modules/react-native-device-info/src/index.ts
index a485044..fe55a43 100644
--- a/node_modules/react-native-device-info/src/index.ts
+++ b/node_modules/react-native-device-info/src/index.ts
@@ -21,7 +21,7 @@ export const getUniqueId = () =>
     defaultValue: 'unknown',
     memoKey: 'uniqueId',
     supportedPlatforms: ['android', 'ios', 'windows'],
-    getter: () => RNDeviceInfo.uniqueId,
+    getter: () => RNDeviceInfo.getUniqueIdSync(),
   });
 
 let uniqueId: string;
@@ -918,6 +918,7 @@ const deviceInfoModule: DeviceInfoModule = {
   getType,
   getTypeSync,
   getUniqueId,
+  getUniqueIdSync: getUniqueId,
   getUsedMemory,
   getUsedMemorySync,
   getUserAgent,
diff --git a/node_modules/react-native-device-info/src/internal/privateTypes.ts b/node_modules/react-native-device-info/src/internal/privateTypes.ts
index 688c6e8..7fbc6d3 100644
--- a/node_modules/react-native-device-info/src/internal/privateTypes.ts
+++ b/node_modules/react-native-device-info/src/internal/privateTypes.ts
@@ -20,7 +20,6 @@ interface NativeConstants {
   model: string;
   systemName: string;
   systemVersion: string;
-  uniqueId: string;
 }
 
 interface HiddenNativeMethods {
@@ -114,6 +113,7 @@ interface ExposedNativeMethods {
   getTotalMemorySync: () => number;
   getType: () => Promise<string>;
   getTypeSync: () => string;
+  getUniqueIdSync: () => string;
   getUsedMemory: () => Promise<number>;
   getUsedMemorySync: () => number;
   getUserAgent: () => Promise<string>;
diff --git a/node_modules/react-native-device-info/windows/code/RNDeviceInfoCPP.h b/node_modules/react-native-device-info/windows/code/RNDeviceInfoCPP.h
index 1fe5953..74a01ce 100644
--- a/node_modules/react-native-device-info/windows/code/RNDeviceInfoCPP.h
+++ b/node_modules/react-native-device-info/windows/code/RNDeviceInfoCPP.h
@@ -32,7 +32,6 @@ namespace winrt::RNDeviceInfoCPP
     REACT_CONSTANT_PROVIDER(constantsViaConstantsProvider);
     void constantsViaConstantsProvider(ReactConstantProvider& provider) noexcept
     {
-      provider.Add(L"uniqueId", getUniqueIdSync());
       provider.Add(L"deviceId", getDeviceIdSync());
       provider.Add(L"serialNumber", getSerialNumberSync());
       provider.Add(L"bundleId", getBundleIdSync());
