diff --git a/node_modules/react-native-remote-svg/index.js b/node_modules/react-native-remote-svg/index.js
index c9893ce..7b4ea86 100644
--- a/node_modules/react-native-remote-svg/index.js
+++ b/node_modules/react-native-remote-svg/index.js
@@ -5,7 +5,7 @@ import SvgImage from './SvgImage';
 
 const MyImage = props => {
   const source = resolveAssetSource(props.source);
-  if (source && (source.uri && source.uri.match('.svg'))) {
+  if (source && (source.uri && source.uri.match('.svg') || source?.type?.includes('svg'))) {
     const style = props.style || {};
     if (source.__packager_asset && typeof style !== 'number') {
       if (!style.width) {
