diff --git a/node_modules/rn-nodeify/browser-browserify.json b/node_modules/rn-nodeify/browser-browserify.json
index 52fa68a..fef79fe 100644
--- a/node_modules/rn-nodeify/browser-browserify.json
+++ b/node_modules/rn-nodeify/browser-browserify.json
@@ -1,10 +1,11 @@
 {
   "zlib": "browserify-zlib",
+  "buffer": "@craftzdog/react-native-buffer",
   "console": "console-browserify",
   "constants": "constants-browserify",
   "crypto": "crypto-browserify",
   "dns": "dns.js",
-  "net": "react-native-tcp",
+  "net": "react-native-tcp-socket",
   "domain": "domain-browser",
   "http": "stream-http",
   "https": "https-browserify",
diff --git a/node_modules/rn-nodeify/browser.json b/node_modules/rn-nodeify/browser.json
index dad26ae..a2af6d8 100644
--- a/node_modules/rn-nodeify/browser.json
+++ b/node_modules/rn-nodeify/browser.json
@@ -1,10 +1,11 @@
 {
   "zlib": "browserify-zlib",
+  "buffer": "@craftzdog/react-native-buffer",
   "console": "console-browserify",
   "constants": "constants-browserify",
-  "crypto": "react-native-crypto",
+  "crypto": "react-native-quick-crypto",
   "dns": "dns.js",
-  "net": "react-native-tcp",
+  "net": "react-native-tcp-socket",
   "domain": "domain-browser",
   "http": "@tradle/react-native-http",
   "https": "https-browserify",
diff --git a/node_modules/rn-nodeify/cmd.js b/node_modules/rn-nodeify/cmd.js
index ae3790c..65efd57 100755
--- a/node_modules/rn-nodeify/cmd.js
+++ b/node_modules/rn-nodeify/cmd.js
@@ -63,7 +63,7 @@ function run () {
   }
 
   if (toShim.indexOf('crypto') !== -1) {
-    toShim.push('react-native-randombytes')
+    toShim.push('react-native-quick-base64')
   }
 
   if (argv.install) {
diff --git a/node_modules/rn-nodeify/shim.js b/node_modules/rn-nodeify/shim.js
index 812d6b4..565a331 100644
--- a/node_modules/rn-nodeify/shim.js
+++ b/node_modules/rn-nodeify/shim.js
@@ -12,7 +12,7 @@ if (typeof process === 'undefined') {
 }
 
 process.browser = false
-if (typeof Buffer === 'undefined') global.Buffer = require('buffer').Buffer
+if (typeof Buffer === 'undefined') global.Buffer = require('@craftzdog/react-native-buffer').Buffer
 
 // global.location = global.location || { port: 80 }
 const isDev = typeof __DEV__ === 'boolean' && __DEV__
diff --git a/node_modules/rn-nodeify/shims-browserify.js b/node_modules/rn-nodeify/shims-browserify.js
index 5952ae2..189260b 100644
--- a/node_modules/rn-nodeify/shims-browserify.js
+++ b/node_modules/rn-nodeify/shims-browserify.js
@@ -17,7 +17,7 @@ module.exports = {
   "punycode": "^1.3.2",
   "querystring-es3": "~0.2.0",
   "react-native-level-fs": "^3.0.0",
-  "react-native-udp": "0.0.12",
+  "react-native-udp": "^4.1.5",
   // "readable-stream": "~1.0.33-1",
   "readable-stream": "^2.0.2",
   "stream-browserify": "^2.0.0",
@@ -27,6 +27,6 @@ module.exports = {
   "url": "~0.10.1",
   "util": "~0.10.1",
   "utp": "0.0.8",
-  "react-native-tcp": "^3.2.1",
+  "react-native-tcp-socket": "^5.6.2",
   "vm-browserify": "~0.0.1"
 }
diff --git a/node_modules/rn-nodeify/shims.js b/node_modules/rn-nodeify/shims.js
index c0e00fa..f3f23f6 100644
--- a/node_modules/rn-nodeify/shims.js
+++ b/node_modules/rn-nodeify/shims.js
@@ -1,12 +1,12 @@
 module.exports = {
   "assert": "^1.1.1",
   "browserify-zlib": "~0.1.4",
-  "buffer": "^4.9.1",
+  "@craftzdog/react-native-buffer": "^6.0.5",
   "inherits": "^2.0.1",
   "console-browserify": "^1.1.0",
   "constants-browserify": "^1.0.0",
-  "react-native-crypto": "^2.1.0",
-  "react-native-randombytes": "^3.0.0",
+  "react-native-quick-crypto": "^0.4.3",
+  "react-native-quick-base64": "^2.0.5",
   "dns.js": "^1.0.1",
   "domain-browser": "^1.1.1",
   "events": "^1.0.0",
@@ -18,7 +18,7 @@ module.exports = {
   "punycode": "^1.2.4",
   "querystring-es3": "~0.2.0",
   "react-native-level-fs": "^3.0.0",
-  "react-native-udp": "^2.1.0",
+  "react-native-udp": "^4.1.5",
   "readable-stream": "1.0.33",
   // "stream-browserify": "substack/stream-browserify#fa56e68",
   "stream-browserify": "^1.0.0",
@@ -27,6 +27,6 @@ module.exports = {
   "tty-browserify": "0.0.0",
   "url": "~0.10.1",
   "util": "~0.10.3",
-  "react-native-tcp": "^3.2.1",
+  "react-native-tcp-socket": "^5.6.2",
   "vm-browserify": "0.0.4"
 }
